src/main/java/com/revature/services/WishlistsService.java

```diff
@@ -0,0 +1,88 @@
+package com.revature.services;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.stereotype.Service;
+
+import com.revature.models.Product;
+import com.revature.models.Wishlist_products;
+import com.revature.models.Wishlists;
+import com.revature.repositories.ProductRepository;
+import com.revature.repositories.Wishlist_productsRepository;
+import com.revature.repositories.WishlistsRepository;
+
+@Service
+public class WishlistsService {
+    
+    private final WishlistsRepository wishlistRepository;
+    private final ProductRepository productRepository;
+    private final Wishlist_productsRepository wishlist_productsRepository;    
+
+    public WishlistsService(WishlistsRepository wishlistRepository, ProductRepository productRepository, Wishlist_productsRepository wishlist_productsRepository) {
+        this.productRepository = productRepository;
+        this.wishlistRepository = wishlistRepository;
+        this.wishlist_productsRepository = wishlist_productsRepository;
+    }
+
+    public List<Wishlists> findAll() {
+        return wishlistRepository.findAll();
+    }
+
+    /**
+     * Returns JSON list of Products based on given user id.
+     * If user does not exist, returns an empty array
+     * @param user_id
+     * @return
+     */
+    public List<Product> findAllFilterById(int user_id) {
```

Comment:
You can avoid having to manually search through all results by adding a custom method to the repository class. There is a special syntax you use for the method name to tell Spring how to build the query. For this method, you would add `public List<Product> findAllByUsers_Id(int userId);` to the repository and call it here.